### PR TITLE
feat(op-challenger): Extend the Challenger Game Filtering to an Allowlist

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Migrates the `--game-address` flag to a string slice allowlist called `--game-allowlist`.

This allows multiple games to be allowed by the challenger to play, with the added bonus
that the api of the challenger binary minimally changes since the allowlist can be
specified with multiple flag instantiations.
e.g. `--game-allowlist 0x0..1 --game--allowlist 0x0..2`

**Metadata**

Fixes CLI-4345

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6926** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6926" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6926?utm_source=stack-comment).

File: op-challenger/config/config.go [hunk 1]
```diff
@@ -24,9 +25,11 @@ var (
 	ErrMissingCannonSnapshotFreq     = errors.New("missing cannon snapshot freq")
 	ErrMissingCannonRollupConfig     = errors.New("missing cannon network or rollup config path")
 	ErrMissingCannonL2Genesis        = errors.New("missing cannon network or l2 genesis path")
-	ErrCannonNetworkAndRollupConfig  = errors.New("only specify one of network or rollup config path")
-	ErrCannonNetworkAndL2Genesis     = errors.New("only specify one of network or l2 genesis path")
-	ErrCannonNetworkUnknown          = errors.New("unknown cannon network")
+	ErrCannonNetworkAndRollupConfig  = errors.New(
```
Comment (trianglesphere) [inline op-challenger/config/config.go]: Shouldn't be re-formatted.

File: op-challenger/flags/flags.go [hunk 2]
```diff
@@ -37,10 +37,10 @@ var (
 		Usage:   "Address of the fault game factory contract.",
 		EnvVars: prefixEnvVars("GAME_FACTORY_ADDRESS"),
 	}
-	GameAddressFlag = &cli.StringFlag{
-		Name:    "game-address",
-		Usage:   "Address of the Fault Game contract.",
-		EnvVars: prefixEnvVars("GAME_ADDRESS"),
+	GameAllowlistFlag = &cli.StringSliceFlag{
+		Name:    "game-allowlist",
+		Usage:   "List of Fault Game contract addresses the challenger is allowed to play.",
```
Comment (trianglesphere) [inline op-challenger/flags/flags.go]: Should leave a comment to the effect that the challenger will play all games if the allowlist is empty.

File: op-challenger/flags/flags.go [hunk 3]
```diff
@@ -63,8 +63,11 @@ var (
 		EnvVars: prefixEnvVars("ALPHABET"),
 	}
 	CannonNetworkFlag = &cli.StringFlag{
-		Name:    "cannon-network",
-		Usage:   fmt.Sprintf("Predefined network selection. Available networks: %s (cannon trace type only)", strings.Join(chaincfg.AvailableNetworks(), ", ")),
+		Name: "cannon-network",
+		Usage: fmt.Sprintf(
```
Comment (trianglesphere) [inline op-challenger/flags/flags.go]: unnecessary diff

**Review by trianglesphere — COMMENTED** [review]
Minor comments, core of the logic looks good.

File: op-challenger/config/config.go (hunk 1 above)
Comment (refcell) [reply op-challenger/config/config.go]: Sorry about this - my editor keeps messing up. Will fix this so I don't accidentally reformat files going forward.

**Review by trianglesphere — APPROVED** [review]
lgtm
